dotcom-rendering/src/web/server/frontToHtml.tsx

```diff
@@ -1,79 +1,145 @@
+}
+
+const generateScriptTags = (
+	scripts: Array<{ src: string; legacy?: boolean } | false>,
+) =>
+	scripts.reduce<string[]>((scriptTags, script) => {
+		if (script === false) return scriptTags;
+
+		let attrs: string;
+		switch (script.legacy) {
+			case true:
+				attrs = 'defer nomodule';
+				break;
+			case false:
+				attrs = 'type="module"';
+				break;
+			default:
+				attrs = 'defer';
+				break;
+		}
+
+		return [
+			...scriptTags,
+			`<script ${attrs} src="${script.src}"></script>`,
+		];
+	}, []);
+
+export const frontToHtml = ({ Front, NAV }: Props): string => {
+	const title = Front.webTitle;
+	const key = 'dcr';
+	const cache = createCache({ key });
+
+	// eslint-disable-next-line @typescript-eslint/unbound-method
+	const { extractCriticalToChunks, constructStyleTagsFromChunks } =
+		createEmotionServer(cache);
+
+	const html = renderToString(
+		<CacheProvider value={cache}>
+			<FrontPage Front={Front} NAV={NAV} />
+		</CacheProvider>,
+	);
+
+	const chunks = extractCriticalToChunks(html);
+	const extractedCss = constructStyleTagsFromChunks(chunks);
+
+	/**
+	 * Preload the following woff2 font files
+	 * TODO: Identify critical fonts to preload
+	 */
+	const fontFiles = [
+		// 'https://assets.guim.co.uk/static/frontend/fonts/guardian-headline/noalts-not-hinted/GHGuardianHeadline-Light.woff2',
+		// 'https://assets.guim.co.uk/static/frontend/fonts/guardian-headline/noalts-not-hinted/GHGuardianHeadline-LightItalic.woff2',
+		'https://assets.guim.co.uk/static/frontend/fonts/guardian-headline/noalts-not-hinted/GHGuardianHeadline-Medium.woff2',
+		'https://assets.guim.co.uk/static/frontend/fonts/guardian-headline/noalts-not-hinted/GHGuardianHeadline-MediumItalic.woff2',
+		'https://assets.guim.co.uk/static/frontend/fonts/guardian-headline/noalts-not-hinted/GHGuardianHeadline-Bold.woff2',
+		'https://assets.guim.co.uk/static/frontend/fonts/guardian-textegyptian/noalts-not-hinted/GuardianTextEgyptian-Regular.woff2',
+		// 'https://assets.guim.co.uk/static/frontend/fonts/guardian-textegyptian/noalts-not-hinted/GuardianTextEgyptian-RegularItalic.woff2',
+		'https://assets.guim.co.uk/static/frontend/fonts/guardian-textegyptian/noalts-not-hinted/GuardianTextEgyptian-Bold.woff2',
+		'https://assets.guim.co.uk/static/frontend/fonts/guardian-textsans/noalts-not-hinted/GuardianTextSans-Regular.woff2',
+		// 'http://assets.guim.co.uk/static/frontend/fonts/guardian-textsans/noalts-not-hinted/GuardianTextSans-RegularItalic.woff2',
+		'https://assets.guim.co.uk/static/frontend/fonts/guardian-textsans/noalts-not-hinted/GuardianTextSans-Bold.woff2',
+	];
+
+	const polyfillIO =
+		'https://assets.guim.co.uk/polyfill.io/v3/polyfill.min.js?rum=0&features=es6,es7,es2017,es2018,es2019,default-3.6,HTMLPictureElement,IntersectionObserver,IntersectionObserverEntry,URLSearchParams,fetch,NodeList.prototype.forEach,navigator.sendBeacon,performance.now,Promise.allSettled&flags=gated&callback=guardianPolyfilled&unknown=polyfill&cacheClear=1';
```

Comment:
Do we want this do be defined independently of Articles?

Reply:
Yeah, I do think things like this should be extracted but I'd like to keep that work separate. I see this duplication work as a way to learn where we both have gaps and also where we have common code. It's good - I think - to be cautious making abstractions as there's potentially cases where we might think we have a common use case but then learn later that there are differences.